pkg/gadget-collection/gadgets/traceloop/gadget.go

```diff
@@ -52,138 +67,298 @@ some differences:
 
 * traceloop uses eBPF instead of ptrace
 * traceloop's tracing granularity is the container instead of a process
-* traceloop's traces are recorded in a fast, in-memory, overwritable ring
+* traceloop's traces are recorded in a fast, in-memory, over writable ring
   buffer like a flight recorder. The tracing could be permanently enabled and
   inspected in case of crash.
 `
 }
 
 func (f *TraceFactory) OutputModesSupported() map[gadgetv1alpha1.TraceOutputMode]struct{} {
 	return map[gadgetv1alpha1.TraceOutputMode]struct{}{
-		gadgetv1alpha1.TraceOutputModeExternalResource: {},
+		gadgetv1alpha1.TraceOutputModeStatus: {},
+		gadgetv1alpha1.TraceOutputModeStream: {},
 	}
 }
 
 func deleteTrace(name string, t interface{}) {
 	trace := t.(*Trace)
-	trace.stop()
+	if trace.started {
+		trace.helpers.Unsubscribe(genPubSubKey(name))
+	}
 }
 
 func (f *TraceFactory) Operations() map[gadgetv1alpha1.Operation]gadgets.TraceOperation {
 	n := func() interface{} {
-		return &Trace{}
+		return &Trace{
+			client:  f.Client,
+			helpers: f.Helpers,
+		}
 	}
 
 	return map[gadgetv1alpha1.Operation]gadgets.TraceOperation{
 		gadgetv1alpha1.OperationStart: {
 			Doc: "Start traceloop",
 			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
-				t := f.LookupOrCreate(name, n).(*Trace)
-				if t.started {
-					trace.Status.State = gadgetv1alpha1.TraceStateStarted
-					return
-				}
-
-				f.mu.Lock()
-				defer f.mu.Unlock()
-
-				if f.started {
-					trace.Status.OperationError = "There is already one traceloop instance running on this node. Please stop it first."
-					return
-				}
-
-				t.Start(trace)
-				f.started = t.started
+				f.LookupOrCreate(name, n).(*Trace).Start(trace)
+			},
+		},
+		gadgetv1alpha1.OperationCollect: {
+			Doc: "Collect traceloop",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				// To overcome Status.Output character limit, we decided to create a
+				// Stream Trace CRD each time we do the collect operation.
+				// Nonetheless, this Trace CRD will use a previously created Trace.
+				// To do so, we use the Parameters["name"] which will contain the name
+				// of the long lived Trace CRD, thus we will be able to get the Trace
+				// and so all the mntNsIDs associated to it.
+				f.LookupOrCreate(trace.Spec.Parameters["name"], n).(*Trace).Collect(trace)
```

Comment:
We need `f.Lookup()` (without 'create') here, otherwise the CLI can add items in the BaseFactory.traces, growing the memory indefinitely without it being released. Or better, the lookup can be done in the Collect() function, so that we can handle errors more easily.

This can be fixed in a future PR.